src/servers/src/proto.rs

```diff
@@ -198,15 +205,19 @@ impl PromTimeSeries {
         }
     }
 
-    fn add_to_table_data(&mut self, table_builders: &mut TablesBuilder) {
+    fn add_to_table_data(&mut self, table_builders: &mut TablesBuilder, is_strict_mode: bool) {
         let label_num = self.labels.len();
         let row_num = self.samples.len();
         let table_data = table_builders.get_or_create_table_builder(
             std::mem::take(&mut self.table_name),
             label_num,
             row_num,
         );
-        table_data.add_labels_and_samples(self.labels.as_slice(), self.samples.as_slice());
+        let _ = table_data.add_labels_and_samples(
```

Comment:
Should we reject requests with invalid utf8 string instead of silently swallow the error?